### ISSUE TITLE
Cant sign into olauncher

### ISSUE THREAD

**Issue by asifashraf80** [olauncher/olauncher#42]
When I try to sign in with Microsoft Olauncher tries to sign in but it gets stuck on a green bar

**Comment by JanGamesHD** [comment]
Can you provide the launcher log?

**Comment by asifashraf80** [comment]
how to get that

**Comment by bigfoot547** [comment]
Is the birthday on your microsoft account set to a date less than 18 years ago? If so, you will need to set up a family to use olauncher.

**Comment by JanGamesHD** [comment]
Do you open the launcher by double clicking on it?

**Comment by asifashraf80** [comment]
yes

**Comment by mortalkombatlegend92** [comment]
Hi, I have the same problem here, but the birthday on my microsoft account is set to over 18 years. Looking through logs when trying to log in with a microsoft account this error occurs.
```

[20:06:27 FATAL]: Unhandled exception in thread Thread[pool-2-thread-2,5,main]
java.lang.IllegalArgumentException: Name and ID cannot both be blank
	at com.mojang.authlib.GameProfile.<init>(GameProfile.java:16) ~[patched.jar:1.7.2_09]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.profileLogIn(MicrosoftUserAuthentication.java:215) ~[patched.jar:1.7.2_09]
	at dev.figboot.olauncher.auth.MicrosoftUserAuthentication.logIn(MicrosoftUserAuthentication.java:133) ~[patched.jar:1.7.2_09]
	at net.minecraft.launcher.ui.popups.login.LogInForm$4.run(LogInForm.java:168) ~[patched.jar:1.7.2_09]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:1.8.0_202]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:1.8.0_202]
	at java.lang.Thread.run(Unknown Source) ~[?:1.8.0_202]
`

**Comment by uptudev** [comment]
@bigfoot547 Here's yet another log for you.

```
[16:17:15 INFO]: OLauncher 1.7.2_09 (through bootstrap 100) started on linux...
[16:17:15 INFO]: Current time is Oct 14, 2023, 4:17:15 PM
[16:17:15 INFO]: System.getProperty('os.name') == 'Linux'
[16:17:15 INFO]: System.getProperty('os.version') == '6.5.6-200.fc38.x86_64'
[16:17:15 INFO]: System.getProperty('os.arch') == 'amd64'
[16:17:15 INFO]: System.getProperty('java.version') == '17.0.8'
[16:17:15 INFO]: System.getProperty('java.vendor') == 'Red Hat, Inc.'
[16:17:15 INFO]: System.getProperty('sun.arch.data.model') == '64'
[16:17:15 INFO]: proxy == DIRECT
[16:17:16 INFO]: Refreshing local version list...
[16:17:16 INFO]: It is not time to query releases yet. Wait another 174 minutes.
[16:17:16 INFO]: Latest release is v1.7.2_09 (ID 110915754).
[16:17:16 INFO]: Refreshing remote version list...
[16:17:16 INFO]: Your java runtime platform is 'linux'.
[16:17:16 INFO]: Finished refreshing java runtime list.
[16:17:16 INFO]: Refresh complete.
[16:17:16 ERROR]: Unexpected exception refreshing profile list
java.lang.NullPointerException: Cannot invoke "net.minecraft.launcher.profile.Profile.getName()" because the return value of "net.minecraft.launcher.profile.ProfileManager.getSelectedProfile()" is null
	at net.minecraft.launcher.Launcher$2.run(Launcher.java:213) ~[patched.jar:1.7.2_09]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[16:17:16 ERROR]: Unhandled exception in executor com.mojang.launcher.updater.ExceptionalThreadPoolExecutor$ExceptionalFutureTask@1e9978d7[Completed exceptionally: java.lang.NullPointerException: Cannot invoke "net.minecraft.launcher.profile.Profile.getName()" because the return value of "net.minecraft.launcher.profile.ProfileManager.getSelectedProfile()" is null]
java.util.concurrent.ExecutionException: java.lang.NullPointerException: Cannot invoke "net.minecraft.launcher.profile.Profile.getName()" because the return value of "net.minecraft.launcher.profile.ProfileManager.getSelectedProfile()" is null
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
	at java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
	at com.mojang.launcher.updater.ExceptionalThreadPoolExecutor$ExceptionalFutureTask.done(ExceptionalThreadPoolExecutor.java:60) ~[patched.jar:1.7.2_09]
	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:381) ~[?:?]
	at java.util.concurrent.FutureTask.setException(FutureTask.java:250) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:269) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.launcher.profile.Profile.getName()" because the return value of "net.minecraft.launcher.profile.ProfileManager.getSelectedProfile()" is null
	at net.minecraft.launcher.profile.ProfileManager.saveProfiles(ProfileManager.java:66) ~[patched.jar:1.7.2_09]
	at net.minecraft.launcher.SwingUserInterface.showLoginPrompt(SwingUserInterface.java:179) ~[patched.jar:1.7.2_09]
	at net.minecraft.launcher.Launcher.ensureLoggedIn(Launcher.java:266) ~[patched.jar:1.7.2_09]
	at net.minecraft.launcher.Launcher$2.run(Launcher.java:242) ~[patched.jar:1.7.2_09]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	... 3 more
[16:17:16 FATAL]: Unhandled exception in thread Thread[AWT-EventQueue-0,6,main]
java.lang.NullPointerException: Cannot invoke "net.minecraft.launcher.profile.Profile.getName()" because the return value of "net.minecraft.launcher.profile.ProfileManager.getSelectedProfile()" is null
	at net.minecraft.launcher.ui.bottombar.ProfileSelectionPanel.populateProfiles(ProfileSelectionPanel.java:81) ~[patched.jar:1.7.2_09]
	at net.minecraft.launcher.ui.bottombar.ProfileSelectionPanel$1.run(ProfileSelectionPanel.java:75) ~[patched.jar:1.7.2_09]
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318) ~[?:?]
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:773) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:720) ~[?:?]
	at java.awt.EventQueue$4.run(EventQueue.java:714) ~[?:?]
	at java.security.AccessController.doPrivileged(AccessController.java:399) ~[?:?]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86) ~[?:?]
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:742) ~[?:?]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203) ~[?:?]
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124) ~[?:?]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113) ~[?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109) ~[?:?]
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) ~[?:?]
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:90) ~[?:?]
```

You're trying to get profile info before generating the profiles necessary upon new installs from what I can tell. Have the launcher initialize and generate a placeholder profile before calling `getName()`, as currently having no profiles is undefined behaviour.

Suffice to say, I strongly doubt this is a login issue, as it seems to be due to a fatal error in the initialization process before one can even login to the launcher entirely, causing the window to stop said process, effectively hanging the window's active elements (while staying responsive).

**Comment by uptudev** [comment]
But yeah tl;dr it seems like the launcher needs to generate some profiles upon initial launch or it crashes.

**Comment by bigfoot547** [comment]
I think the log you posted is the result of a different issue than the one the author is experiencing. The only way that `getSelectedProfile()` could return null is if there was an invalid (null) profile read from the launcher profiles, and it happened to be the selected profile. `getSelectedProfile()` will create a fallback profile in any other circumstance.

**Comment by uptudev** [comment]
There are no profiles yet. This is upon initial running of the JAR, having not run the old launcher before on the system, so the creation of the fallback profile seems to be faulty in some regard.

Shouldn't a fallback be created specifically if a null profile is read to prevent this sort of behaviour from being passed through?

Anyways, I can make this into a new issue if you'd prefer that I not clog up this thread. I noticed it to be a direct parallel to @mortalkombatlegend92's log, so figured it was worth looking into.

**Comment by uptudev** [comment]
Is there a default location I could check to clear any profiles in the directory to try to force the fallback? If this launcher uses the default `.minecraft` folder, it could be a result of running it after the new Microsoft launcher, as idrk if they have conflicting profile syntaces.

**Comment by bigfoot547** [comment]
There's no need to make a new issue, the bug you describe is already tracked in #21 (though I may change the title to something more descriptive). The issue really is that the `name` attribute in the profile does does not necessarily match the actual key of the profile in `launcher_profiles.json`. The fallback fails when this is the case.

**Comment by uptudev** [comment]
Found the bug from that, thanks.

The Microsoft launcher initializes that JSON with nothing in the name field. Anyone running after having installed the new installer (or installed in parallel) may face this error.

```json
{
  "profiles" : {
    "646953ffb64d47e2efc25b103fa52e70" : {
      "created" : "1970-01-02T00:00:00.000Z",
      "icon" : "Grass",
      "lastUsed" : "1970-01-02T00:00:00.000Z",
      "lastVersionId" : "latest-release",
      "name" : "",  //HERE
      "type" : "latest-release"
    },
    "c0514daef10423875c0631616669bb8f" : {
      "created" : "1970-01-01T00:00:00.000Z",
      "icon" : "Dirt",
      "lastUsed" : "1970-01-01T00:00:00.000Z",
      "lastVersionId" : "latest-snapshot",
      "name" : "",  //AND HERE
      "type" : "latest-snapshot"
    }
  },
  "settings" : {
    "crashAssistance" : false,
    "enableAdvanced" : false,
    "enableAnalytics" : true,
    "enableHistorical" : false,
    "enableReleases" : true,
    "enableSnapshots" : false,
    "keepLauncherOpen" : false,
    "profileSorting" : "ByLastPlayed",
    "showGameLog" : false,
    "showMenu" : false,
    "soundOn" : false
  },
  "version" : 3
}
```

Should be an easy fix on my end to confirm. I'll report back shortly.

**Comment by uptudev** [comment]
Solution is to delete all of the "latest" profiles generated by the new launcher. Thanks for the support!

**Comment by bigfoot547** [comment]
I think I'll release a hotfix to properly migrate the profiles. I seem to have gotten something working.

**Comment by mortalkombatlegend92** [comment]
Hi, as if the recent update I still get the "Name and ID cannot both be blank" error I mentioned earlier, I've seen in an old bug issue someone had the same error as me but theirs randomly fixed itself from what I read.

**Comment by mortalkombatlegend92** [comment]
Ok never mind, it looks like I am outrageously dumb and was logging into the wrong account, sorry for the trouble.

**Comment by bigfoot547** [comment]
That'd do it. No worries, enjoy the launcher :)
